server/netty-server/src/main/scala/sttp/tapir/server/netty/internal/NettyConnectionCounter.scala

```diff
@@ -0,0 +1,20 @@
+package sttp.tapir.server.netty.internal
+
+import io.netty.channel.ChannelHandler.Sharable
+import io.netty.channel.{ChannelHandlerContext, ChannelInboundHandlerAdapter}
+
+import java.util.concurrent.atomic.AtomicInteger
+
+@Sharable case class NettyConnectionCounter(maxConnections: Int) extends ChannelInboundHandlerAdapter {
```

Comment:
do we need the annotation?

Reply:
Yes, according documentation with this annotation in Netty class can be safely shared among multiple connections and it prevent race condition.